vignettes/MsBackend.Rmd

```diff
@@ -563,35 +563,39 @@ additionally available variables and the `columns` parameter of the
 (in addition to the required `"mz"` and `"intensity"` variables).
 
 
-### `[`
-
-The `[` method allows to subset `MsBackend` objects. This operation is expected
-to reduce a `MsBackend` object to the selected spectra. The method should
-support to subset by indices or logical vectors and should also support
-duplicating elements (i.e. when duplicated indices are used) as well as to
-subset in arbitrary order. An error should be thrown if indices are out of
-bounds, but the method should also support returning an empty backend with
-`[integer()]`. Note that the `MsCoreUtils::i2index` function can be used to
+### `extractByIndex()` and `[`
+
+The `extractByIndex()` and `[` methods allows to subset `MsBackend` objects.
+This operation is expected to reduce a `MsBackend` object to the selected
+spectra. These methods must also support duplication (e.g. `[c(1, 1, 1)]` and
+extraction in any arbitrary order (e.g. `[c(3, 1, 5, 3)]`). While both methods
+subset the object, `extractByIndex()` only supports to subset with an `integer`
+index, while `[`, to be compliant with the base R implementation, should support
+to subset by indices or logical vectors. An error should be thrown if indices
+are out of bounds, but the method should also support returning an empty backend
+with `[integer()]`. Note that the `MsCoreUtils::i2index` function can be used to
 check for correct input (and convert the input to an `integer` index).
 
-Below we implement a possible `[` for our test backend class. We ignore the
-parameters `j` from the definition of the `[` generic, since we treat our data
-to be one-dimensional (with each spectrum being one element).
+The `extractByIndex()` method is used by the data operation and analysis methods
```

Comment:
just so I understand, from now on when we need to implement some code in Spectra we have to use `extractByIndex()` instead of `[` ?

Reply:
we can use both - but it would be *better* to use `extractByIndex`.